### PR TITLE
[Docs] Use Pure Dataclass In Example

### PR DESCRIPTION
**Pull request by Future-Outlier**
## Docs link
flyte dataclass: https://flyte--5829.org.readthedocs.build/en/5829/user_guide/data_types_and_io/dataclass.html

### CODE REVIEW

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 1]
```diff
@@ -25,15 +25,21 @@ To clone and run the example code on this page, see the [Flytesnacks repo][flyte
 
 To begin, import the necessary dependencies:
 
-```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/69dbe4840031a85d79d9ded25f80397c6834752d/examples/data_types_and_io/data_types_and_io/dataclass.py
+```{rli} https://raw.githubusercontent.com/flyteorg/flytesnacks/cfb5ea3b0d0502ef7df1f2e14f4a0d9b78250b6a/examples/data_types_and_io/data_types_and_io/dataclass.py
+:caption: data_types_and_io/dataclass.py
+:lines: 1-9
+```
+
+Build your custom image by ImageSpec:
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:33]:
```suggestion
Build your custom image with ImageSpec:
```

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 2]
```diff
@@ -12,11 +12,11 @@ Flytekit uses the [Mashumaro library](https://github.com/Fatal1ty/mashumaro)
 to serialize and deserialize dataclasses.
 
 :::{important}
-If you're using Flytekit version below v1.10, you'll need to decorate with `@dataclass_json` using
-`from dataclasses_json import dataclass_json` instead of inheriting from Mashumaro's `DataClassJSONMixin`.
-
-If you're using Flytekit version >= v1.11.1, you don't need to decorate with `@dataclass_json` or
+If you're using Flytekit version below v1.11.1, your dataclass will need to decorate with `@dataclass_json` or
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:15]: Just to clarify, this means that if you are using Flytekit < 1.11.1, you either need to decorate your dataclass with `@dataclass_json` or create a class that inherits `DataClassJSONMixin`, but that you don't need to do both? If you inherit from `DataClassJSONMixin`, do you need to do `from mashumaro.mixins.json import DataClassJSONMixin` as well?

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 3]
```diff
@@ -12,11 +12,11 @@ Flytekit uses the [Mashumaro library](https://github.com/Fatal1ty/mashumaro)
...
 inherit from Mashumaro's `DataClassJSONMixin`.
+
+If you're using Flytekit version below v1.10, you'll need to decorate with `@dataclass_json` using
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:18]:
```suggestion
If you're using Flytekit version less than v1.10, you'll need to add `from dataclasses_json import dataclass_json` to your imports  and decorate your dataclass with `@dataclass_json`.
```

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 4]
```diff
@@ -12,11 +12,11 @@ Flytekit uses the [Mashumaro library](https://github.com/Fatal1ty/mashumaro)
...
+`from dataclasses_json import dataclass_json`
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:19]:
```suggestion
```

**Review by neverett — CHANGES_REQUESTED** [review]
I left a few questions and suggestions -- let me know what you think!

File: docs/user_guide/data_types_and_io/dataclass.md (hunk 2 above)
Comment (Future-Outlier) [reply docs/user_guide/data_types_and_io/dataclass.md:15]: yes!

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 5]
```diff
@@ -12,11 +12,10 @@ Flytekit uses the [Mashumaro library](https://github.com/Fatal1ty/mashumaro)
 to serialize and deserialize dataclasses.
 
 :::{important}
-If you're using Flytekit version below v1.10, you'll need to decorate with `@dataclass_json` using
-`from dataclasses_json import dataclass_json` instead of inheriting from Mashumaro's `DataClassJSONMixin`.
-
-If you're using Flytekit version >= v1.11.1, you don't need to decorate with `@dataclass_json` or
+If you're using Flytekit version below v1.11.1, your dataclass will need to decorate with `@dataclass_json` or
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:15]:
```suggestion
If you're using Flytekit version greater than v1.10 and less than v1.11.1, you will need add `from mashumaro.mixins.json import DataClassJSONMixin` to your imports and either decorate your dataclass with `@dataclass_json` or create a class that inherits from `DataClassJSONMixin`. 
```

Comment (neverett) [reply docs/user_guide/data_types_and_io/dataclass.md:15]: Let me know if this is accurate!

Comment (Future-Outlier) [reply docs/user_guide/data_types_and_io/dataclass.md:15]: it's not, but I can update it

File: docs/user_guide/data_types_and_io/dataclass.md [hunk 6]
```diff
@@ -12,11 +12,7 @@ Flytekit uses the [Mashumaro library](https://github.com/Fatal1ty/mashumaro)
 to serialize and deserialize dataclasses.
 
 :::{important}
-If you're using Flytekit version below v1.10, you'll need to decorate with `@dataclass_json` using
-`from dataclasses_json import dataclass_json` instead of inheriting from Mashumaro's `DataClassJSONMixin`.
-
-If you're using Flytekit version >= v1.11.1, you don't need to decorate with `@dataclass_json` or
-inherit from Mashumaro's `DataClassJSONMixin`.
+If you're using Flytekit version below v1.11.1, your dataclass will need to add `from dataclasses_json import dataclass_json` and decorate with `@dataclass_json`.
```
Comment (neverett) [inline docs/user_guide/data_types_and_io/dataclass.md:15]:
```suggestion
If you're using Flytekit version below v1.11.1, you will need to add `from dataclasses_json import dataclass_json` to your imports and decorate your dataclass with `@dataclass_json`.
```

**Review by neverett — APPROVED** [review]

**Review by neverett — APPROVED** [review]

**Pull request merged**